src/gosling-track/gosling-track.ts

```diff
@@ -836,100 +832,112 @@ function GoslingTrack(HGC: import('@higlass/types').HGC, ...args: any[]): any {
+                }
+                return tileInfo.goslingModels;
             }
 
-            if (!tile.gos.tilePos) {
-                // we do not have this information ready yet, so we cannot get tileX
-                return;
+            if (!tile.tileData.tilePos) {
+                // we do not have this information ready yet, i.e., cannot calculate `tileX`
+                return [];
             }
 
-            // Single tile can contain multiple gosling models if multiple tracks are superposed.
-            tile.goslingModels = [];
-
-            const spec = JSON.parse(JSON.stringify(this.options.spec));
+            const tileInfo = this.processedTileInfo[tile.tileId];
+            const specCopy = JSON.parse(JSON.stringify(this.options.spec));
+            // interactive brushes are drawn by another plugin track, called `gosling-brush`
+            const resolvedTracks = resolveSuperposedTracks(specCopy).filter(t => t.mark !== 'brush');
 
-            const [trackWidth, trackHeight] = this.dimensions; // actual size of a track
-
-            resolveSuperposedTracks(spec).forEach(resolved => {
-                if (resolved.mark === 'brush') {
-                    // interactive brushes are drawn by another plugin track, called `gosling-brush`
-                    return;
-                }
-
-                if (!tile.gos.tabularData) {
-                    // If the data is not already stored in a tabular form, convert them.
+            /* generate tabular data */
+            resolvedTracks.forEach(resolvedSpec => {
+                if ('tabularData' in tile.tileData) {
+                    // some data fetchers directly generates `tabularData`
+                    tileInfo.tabularData = tile.tileData.tabularData;
+                } else {
+                    // generated tabular data
                     const { tileX, tileY, tileWidth, tileHeight } = this.getTilePosAndDimensions(
-                        tile.gos.zoomLevel,
-                        tile.gos.tilePos,
+                        tile.tileData.zoomLevel,
+                        tile.tileData.tilePos!,
                         this.tilesetInfo.bins_per_dimension || this.tilesetInfo?.tile_size
                     );
 
-                    tile.gos.tabularData = getTabularData(resolved, {
-                        ...tile.gos,
+                    const sparse = 'length' in tile.tileData ? Array.from(tile.tileData) : [];
+                    const tabularData = getTabularData(resolvedSpec, {
+                        ...tile.tileData,
+                        sparse,
                         tileX,
                         tileY,
                         tileWidth,
                         tileHeight,
                         tileSize: this.tileSize
                     });
+                    if (tabularData) {
+                        tileInfo.tabularData = tabularData;
+                    }
                 }
+            });
```

Comment:
Am I correct that `tileInfo.tabularData` is continuously re-written if `resolvedTracks.length >= 1`?